### PR TITLE
chore(doc): config file document

### PR DESCRIPTION
**Pull request by jayzhang**
#1238

### CODE REVIEW

File: docs/fx-core/config.md [hunk 1]
```diff
@@ -15,4 +69,96 @@ botId|string|The AAD App client id generated during provision which is paired wi
...
+This section is to describe configuration items in `fx-resource-azure-sql` section of `.fx/env.default.json`.
+
+Config Name | Config Type | Description
+------|------|------
+sqlEndpoint	|string|	The sql server endpoint.  '${sqlServer}.database.windows.net'
+databaseName|	string|	The created database name
+userName|	string|	account login for local debug
+password|	string|	account password for local debug
+
+## Configuration schema for Azure Function
```
Comment (henzhang-ms) [inline docs/fx-core/config.md:104]: please change `Azure Function` to `Azure Functions`, thanks, that's the correct term.

File: docs/fx-core/config.md [hunk 2]
```diff
@@ -15,4 +69,96 @@ botId|string|The AAD App client id generated during provision which is paired wi
...
+
+This section is to describe configuration items in `fx-resource-function` section of `.fx/env.default.json`.
+
+Config Name | Config Type | Description
+------|------|------
+defaultFunctionName	|string	|The default function name scaffolded
+functionAppName	|string	|Output. The functions app name.
+functionEndpoint|	string|	Output. The functions app endpoint. `https://${functionAppName}.azurewebsites.net`
```
Comment (henzhang-ms) [inline docs/fx-core/config.md:112]: and the space alignment is inconsistent

Comment (henzhang-ms) [reply docs/fx-core/config.md:112]: and it is `function app` not `functions app`

File: docs/fx-core/config.md [hunk 3]
```diff
@@ -15,4 +69,98 @@ botId|string|The AAD App client id generated during provision which is paired wi
...
+## Configuration schema for Azure SQL
+
+This section is to describe configuration items in `fx-resource-azure-sql` section of `.fx/env.default.json`.
+
+Config Name | Config Type | Description
+------|------|------
+sqlEndpoint	|string|	The sql server endpoint.  '${sqlServer}.database.windows.net'
+databaseName|	string|	The created database name
+userName|	string|	account login for local debug
+password|	string|	account password for local debug
```
Comment (xzf0587) [inline docs/fx-core/config.md]: The username and password are sensitive and may not be set in config. It is better to remove them here.

**Review by LongOddCode — APPROVED** [review]

File: docs/fx-core/config.md [hunk 4]
```diff
@@ -15,4 +69,96 @@ botId|string|The AAD App client id generated during provision which is paired wi
...
+
+## Configuration schema for Azure Functions
+
+This section is to describe configuration items in `fx-resource-function` section of `.fx/env.default.json`.
+
+Config Name | Config Type | Description
+------|------|------
+defaultFunctionName|string|The default function name scaffolded
+functionAppName|string|Output. The function app name.
+functionEndpoint|string|Output. The function app endpoint. `https://${functionAppName}.azurewebsites.net`
```
Comment (henzhang-ms) [inline docs/fx-core/config.md:112]: Sorry, please help add the tow config
```
storageAccountName|string|Output. The name of the Azure Storage Account used by the function app.
appServicePlanName|string|Output. The name of the Azure App Service Plan used by the function app.
```

Comment (henzhang-ms) [reply docs/fx-core/config.md:112]: They are not in contract, but will be exposed to user. I saw other plugin put the internal config in schema. So I think function also needs them.

**Review by henzhang-ms — APPROVED** [review]

**Pull request merged**